tests/Component.Tests/EventDispatcherTests.cs

```diff
@@ -0,0 +1,47 @@
+﻿using Component.Tests.Fakes;
+using Components;
+using FluentAssertions;
+using Microsoft.Framework.DependencyInjection;
+using System;
+using System.Linq;
+using Xunit;
+
+namespace Component.Tests
+{
+    public class EventDispatcherTests
+    {
+        static IServiceProvider CreateProvider() => new ServiceCollection().AddInMemoryEventStream().BuildServiceProvider();
+        static EventStream CreateEventStream(IServiceProvider serviceProvider = null) => (serviceProvider ?? CreateProvider()).GetRequiredService<EventStream>();
+        static EventDispatcher CreateDispatcher(IServiceProvider serviceProvider = null) => (serviceProvider ?? CreateProvider()).GetRequiredService<EventDispatcher>();
+
+        [Fact]
+        public static void GivenANullEntityThrowArgumentNullException()
```

Comment:
Either follow the naming convention of the existing tests (preferably) or change all the existing tests to follow this format. Can let @herecydev / @danielcirket / @mattridgway decide the format they prefer?